ts/Stock/Indicators/TrendLine/TrendLineIndicator.ts

```diff
@@ -107,48 +107,55 @@ class TrendLineIndicator extends SMAIndicator {
         series: TLinkedSeries,
         params: TrendLineParamsOptions
     ): IndicatorValuesObject<TLinkedSeries> {
-        const xVal: Array<number> = (series.xData as any),
+        const orgXVal: Array<number> = (series.xData as any),
             yVal: Array<Array<number>> = (series.yData as any),
+            xVal: Array<number> = [],
             LR: Array<Array<number>> = [],
             xData: Array<number> = [],
             yData: Array<number> = [],
-            xValLength: number = xVal.length,
-            index: number = (params.index as any);
-
-        let sumX = (xValLength - 1) * xValLength / 2,
-            sumY = 0,
-            sumXY = 0,
-            sumX2 =
-                ((xValLength - 1) * (xValLength) * (2 * xValLength - 1)) / 6,
-            alpha: number,
-            i: number,
-            y: number;
-
-        // Get sums:
-        for (i = 0; i < xValLength; i++) {
-            y = isArray(yVal[i]) ? yVal[i][index] : (yVal[i] as any);
-            sumY += y;
-            sumXY += i * y;
+            index: number = params.index || 3;
+
+        let numerator = 0,
+            denominator = 0,
+            xValSum = 0,
+            yValSum = 0,
+            counter = 0;
+
+        // Create an array of consecutive xValues, (don't remove duplicates)
+        for (let i = 0; i < orgXVal.length; i++) {
+            if (i === 0 || orgXVal[i] !== orgXVal[i - 1]) {
+                counter++;
```

Comment:
Just double checking: it means we start the array from x=1, is this expected? Previous solution started from x=0.

Reply:
Yes, exactly - by default HC xData array starts from 1 (if no value is given), and that what we want to achieve here.